### PR TITLE
Drop support for Python 3.8

### PR DESCRIPTION
**Pull request by raulcd**
This PR drops support for Python 3.8 as successfully voted on the Mailing List here: https://lists.apache.org/thread/50jjtdyfovwqbw8mp5m6rfgmmbxv7qcr

Closes https://github.com/apache/iceberg-python/issues/1121

### CODE REVIEW

**Comment by raulcd** [comment]
The integration job uses Ubuntu 20.04 which will be EOL on April 2025. Ubuntu 20.04 uses Python 3.8 as default, let me know if you'd prefer to use Ubuntu 22.04 on that job instead of installing Python 3.9 as I've done on the PR.
For reference, on Apache Arrow we have decided to stop testing pyarrow for Ubuntu 20.04 and have updated to 22.04/24.04 even though we still build Arrow C++ for Ubuntu 20.04 and will deprecate closer to EOL.

**Comment by sungwy** [comment]
Hi @raulcd thank you very much for putting together this PR. I'm +1 for using ubuntu-22.04 for the integration test to align with what arrow is doing. This will also align the machine with the python CI github workflow, which is actually already running on ubuntu-22.04.

**Comment by raulcd** [comment]
Thanks @sungwy, I've update the job to Ubuntu 22.04. CI is successful.

File: mkdocs/docs/contributing.md [hunk 1]
```diff
@@ -202,7 +202,7 @@ Deprecated in 0.1.0, will be removed in 0.2.0. The old_property is deprecated. P
 
 For the type annotation the types from the `Typing` package are used.
 
-PyIceberg offers support from Python 3.8 onwards, we can't use the [type hints from the standard collections](https://peps.python.org/pep-0585/).
+PyIceberg offers support from Python 3.9 onwards, we can't use the [type hints from the standard collections](https://peps.python.org/pep-0585/).
```
Comment (kevinjqliu) [inline mkdocs/docs/contributing.md:205]: nit: add a paragraph about dropping 3.8 support at version 0.x (whatever we release this commit as)

File: .github/workflows/python-release.yml [hunk 2]
```diff
@@ -44,7 +44,7 @@ jobs:
       - uses: actions/setup-python@v5
         with:
           python-version: |
-            3.8
+            3.9
             3.11
```
Comment (kevinjqliu) [inline .github/workflows/python-release.yml:48]: nit, can we add 3.10 here too

**Review by kevinjqliu — COMMENTED** [review]
Thanks for the contribution! I've added a few nit comments. 

We should figure out where we want to release this PR and mark which release we officially drop python 3.8

File: .github/workflows/python-release.yml (hunk 2 above)
Comment (raulcd) [reply .github/workflows/python-release.yml:49]: Done!

File: mkdocs/docs/contributing.md (hunk 1 above)
Comment (kevinjqliu) [reply mkdocs/docs/contributing.md:205]: ah sorry about the confusion, i didnt see this is the `contributing` doc. in that case, this isn't needed.

i thought this was the docs on how to use pyiceberg, in which case we should mention that 3.8 support is dropped in this next release

**Review by HonahX — APPROVED** [review]
LGTM! @raulcd Thanks for the contribution!

**Review by sungwy — APPROVED** [review]

**Comment by kevinjqliu** [comment]
Merging, since it was already voted to drop python 3.8 in the pyiceberg 0.8.0 release (next release)

**Comment by kevinjqliu** [comment]
Thank you @raulcd for the contribution!

**Comment by raulcd** [comment]
> Thank you @raulcd for the contribution!

Thanks @kevinjqliu @sungwy @HonahX for approving the PR! I plan to keep contributing :)

**Pull request merged**